pytorch_lightning/callbacks/progress/tqdm_progress.py

```diff
@@ -273,62 +291,75 @@ def on_train_epoch_start(self, trainer: "pl.Trainer", *_: Any) -> None:
         self.main_progress_bar.set_description(f"Epoch {trainer.current_epoch}")
 
     def on_train_batch_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule", *_: Any) -> None:
-        if self._should_update(self.train_batch_idx):
+        if self._should_update(self.train_batch_idx, self.total_train_batches):
             _update_n(self.main_progress_bar, self.train_batch_idx + self._val_processed)
             self.main_progress_bar.set_postfix(self.get_metrics(trainer, pl_module))
 
     def on_train_epoch_end(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
-        _update_n(self.main_progress_bar, self.train_batch_idx + self._val_processed)
```

Comment:
Why remove this?

We did not need to pass the `total` to `_update_n` before because the epoch end hooks did update unconditionally.

Reply:
the progress bar is updated only in batch-level hooks. check the update in `_should_update`.
I am trying to bring rich and tqdm progress bar API closer. This was done just to make sure that main progress bar exits with proper counter.

Reply:
This was done on purpose: https://github.com/PyTorchLightning/pytorch-lightning/pull/11069/files#r769739944 because the logger connector state will be different in these two hooks.

> I am trying to bring rich and tqdm progress bar API closer

In any case, I think rich should also update in `epoch_end`.
But there's no need to make both implementations exactly equal, they will have their natural differences as they use different libraries.

Reply:
I tried a few examples.. seems to be working fine.. but maybe I missed something?
can you share a trainer config where it will fail?